indico/modules/events/registration/schemas.py

```diff
@@ -33,3 +39,95 @@ def sort_list(self, data, many, **kwargs):
+                  'registration_date', 'registration_data')
+
+    regform_id = fields.Int(attribute='registration_form_id')
+    full_name = fields.Str(attribute='display_full_name')
+    state = fields.Enum(RegistrationState)
+    checkin_secret = fields.UUID(attribute='ticket_uuid')
+    formatted_price = fields.Function(lambda reg: reg.render_price())
+    tags = fields.Function(lambda reg: sorted(t.title for t in reg.tags))
+    registration_date = fields.DateTime(attribute='submitted_dt')
+    registration_data = fields.Method('_get_registration_data')
+
+    def _get_filenames(self, registration):
+        """Extract filenames from file fields."""
+        return {r.field_data.field.id: r.filename
+                for r in registration.data
+                if r.field_data.field.field_impl.is_file_field and r.storage_file_id is not None}
+
+    def _get_registration_data(self, registration):
+        regform = registration.registration_form
+        form_data = get_flat_section_submission_data(regform, registration=registration, management=True)
+        reg_data = get_form_registration_data(regform, registration, management=True)
+        filenames = self._get_filenames(registration)
+        data = {}
+
+        for section_id, section in form_data['sections'].items():
+            data[section_id] = {
+                'id': section_id,
+                'position': section['position'],
+                'title': section['title'],
+                'description': section['description'],
+                'fields': []
+            }
+
+        for field_id, field in form_data['items'].items():
+            section = data[field['sectionId']]
+            field_data = {
+                'id': field_id,
+                'position': field['position'],
+                'title': field['title'],
+                'description': field['description'],
+                'inputType': field['inputType'],
+                'data': reg_data[field['htmlName']],
+                'defaultValue': field['defaultValue'],
+            }
+            if 'price' in field:
+                field_data['price'] = field['price']
+            if 'choices' in field:
+                field_data['choices'] = field['choices']
+            # File field stores the uuid as data which is not helpful.
+            # We want to show the filename instead.
+            if field_id in filenames:
+                field_data['data'] = filenames[field_id]
+            section['fields'].append(field_data)
+
+        # Sort sections and fields based on 'position'
+        data = list(data.values())
+        data.sort(key=itemgetter('position'))
+        for sections in data:
+            sections['fields'].sort(key=lambda f: f['position'])
+
+        # Remove the 'position' key
+        for section in data:
+            del section['position']
+            for field in section['fields']:
+                del field['position']
```

Comment:
Instead of sorting and removing the key, I think you could also sort before generating the `data` dict since nowadays dicts are always sorted by insertion order.